src/main/kotlin/app/revanced/patches/all/ads/blocklist/AndroidAdsBlocklist.kt

```diff
@@ -0,0 +1,31 @@
+package app.revanced.patches.all.ads.blocklist
+
+import org.jf.dexlib2.Opcode
+
+val blockUrls: List<String> = listOf(
+    "adservice.google.com",
+    "g.doubleclick.net",
+    "googleadservices.com",
+    "amazon-adsystem.com",
+)
+
+val replaceUrlsWith: String = "https://example.com"
+
+val stringOpcodes: List<Opcode> = listOf(
+    Opcode.CONST_STRING
+)
+
+val blockInvokes: List<String> = listOf(
+    "loadAd",
+    "AdListener",
+    "AdLoad",
+    "AdView",
+)
+
+val invokeOpcodes: List<Opcode> = listOf(
+    Opcode.INVOKE_VIRTUAL,
+    Opcode.INVOKE_SUPER,
+    Opcode.INVOKE_DIRECT,
+    Opcode.INVOKE_STATIC,
+    Opcode.INVOKE_INTERFACE
+)
```

Comment:
You can check if the current instruction is a ReferenceInstruction and if the reference is a method reference if you want to take a shortcut, may be less desired though than just checking the opcode.